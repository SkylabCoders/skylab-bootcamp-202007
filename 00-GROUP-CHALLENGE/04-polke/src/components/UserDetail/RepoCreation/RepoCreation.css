.repocreation__form {
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: flex-start;
	background-color: var(--form-color-background);
	border-radius: 7px;
	padding: 16px;
	position: absolute;
	z-index: 10;
	margin-bottom: 64px;
	box-shadow: 2px 2px #888888;
}

.repocreation__form:before {
	content: '';

	position: absolute;
	height: 0px;
	width: 0px;
	top: 50%;
	left: -7%;
	border-width: 15px;
	border-color: transparent var(--form-color-background) transparent transparent;
	border-style: solid;
}

.form__repo-info {
	display: flex;
	flex-direction: column;
	align-items: center;
}

.form-section {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: flex-end;
	width: 100%;
	padding-bottom: 24px;
}

.form_access-container {
	display: flex;
	flex-direction: column;
}

.form_access-container p {
	display: flex;
	flex-direction: column;
	margin-bottom: 2px;
}

.form__repo-access {
	border-radius: 5px;
	margin-left: 100px;
}

.form__text-input {
	padding: 5px;
	margin-right: 5px;
	border-radius: 5px;
	border: 0.3px solid gray;
}

.form__readme {
	margin: 10px;
}

.form__bottom {
	display: flex;
	flex-direction: row;
	justify-content: space-around;

	width: 100%;
}

.repo__button {
	background-color: var(--bg-color-button);
	border-radius: 5px;
	border: none;
	color: whitesmoke;
	width: 25%;
	cursor: pointer;
}
