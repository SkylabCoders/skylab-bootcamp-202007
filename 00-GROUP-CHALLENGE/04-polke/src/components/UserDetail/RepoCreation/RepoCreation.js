import React, { useState } from 'react';
import landingStore from '../../../stores/landingStore';
import './RepoCreation.css';
import '../../../shared/generalStyles.css';
import { createRepo } from '../../../actions/userDetailActions';

function RepoCreation() {
	const [gitHubAccessToken] = useState(landingStore.getGitHubAccessToken());

	return (
		<div className="repocreation__container">
			<form className="repocreation__form">
				<div className="form__repo-info">
					<div className="form-section">
						<input
							type="text"
							name="repo-name"
							className="form__repo-name form__text-input"
							placeholder="Repo name..."
						/>
						<div className="form_access-container">
							<p> Repo Access: </p>
							<select className="form__repo-access" name="repo-access">
								<option value="false" selected>
									Public
								</option>
								<option value="true">Private</option>
							</select>
						</div>
					</div>
					<div className="form-section">
						<textarea
							name="repo-desc"
							className="form__repo-desc form__text-input"
							placeholder="Repo description..."
							rows="2"
							cols="40"
						></textarea>
					</div>
				</div>

				<div className="form__bottom">
					<div>
						<label for="repo-readme"> Add README.md</label>
						<input
							type="checkbox"
							className="form__readme"
							name="repo-readme"
							value="readme"
						/>
					</div>
					<button
						className="repo__button"
						onClick={(event) => {
							event.preventDefault();
							const name = document.getElementsByClassName('form__repo-name')[0]
								.value;
							const access = false;
							const desc = document.getElementsByClassName('form__repo-desc')[0]
								.value;
							const readme = false;
							createRepo(name, access, desc, readme, gitHubAccessToken);
						}}
					>
						Create
					</button>
				</div>
			</form>
		</div>
	);
}

export default RepoCreation;
