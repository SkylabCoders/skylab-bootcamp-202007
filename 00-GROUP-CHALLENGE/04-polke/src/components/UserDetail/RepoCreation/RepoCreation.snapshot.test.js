import renderer from 'react-test-renderer';
import RepoCreation from './RepoCreation';
import React from 'react';

describe('RepoCreation snapshot', () => {
	const tree = renderer.create(<RepoCreation />);
	it('should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
