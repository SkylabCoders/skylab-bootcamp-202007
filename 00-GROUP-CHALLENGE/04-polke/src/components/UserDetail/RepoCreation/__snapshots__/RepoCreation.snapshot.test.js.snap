// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCreation snapshot should match 1`] = `
<div
  className="repocreation__container"
>
  <form
    className="repocreation__form"
  >
    <div
      className="form__repo-info"
    >
      <div
        className="form-section"
      >
        <input
          className="form__repo-name form__text-input"
          name="repo-name"
          placeholder="Repo name..."
          type="text"
        />
        <div
          className="form_access-container"
        >
          <p>
             Repo Access: 
          </p>
          <select
            className="form__repo-access"
            name="repo-access"
          >
            <option
              selected={true}
              value="false"
            >
              Public
            </option>
            <option
              value="true"
            >
              Private
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-section"
      >
        <textarea
          className="form__repo-desc form__text-input"
          cols="40"
          name="repo-desc"
          placeholder="Repo description..."
          rows="2"
        />
      </div>
    </div>
    <div
      className="form__bottom"
    >
      <div>
        <label
          for="repo-readme"
        >
           Add README.md
        </label>
        <input
          className="form__readme"
          name="repo-readme"
          type="checkbox"
          value="readme"
        />
      </div>
      <button
        className="repo__button"
        onClick={[Function]}
      >
        Create
      </button>
    </div>
  </form>
</div>
`;
